tpm/tpm.go

```diff
@@ -325,12 +325,7 @@ func zeroBytes(b []byte) {
 }
 
 // Seal encrypts data against a given locality and PCRs and returns the sealed data.
-func Seal(rw io.ReadWriter, locality byte, pcrs []int, data []byte, srkAuth []byte) ([]byte, error) {
-	pcrInfo, err := newPCRInfoLong(rw, locality, pcrs)
-	if err != nil {
-		return nil, err
-	}
-
+func Seal(rw io.ReadWriter, locality byte, pcrs []int, customPcrs *PcrInfoLong, data []byte, srkAuth []byte) ([]byte, error) {
```

Comment:
I'm not sure this is the right API, but I might not be understanding the intended usage (which is why I'm asking for more details). I suspect that there are two non-overlapping uses of Seal: one that seals to the current PCR values, and one that seals to known PCR values that are not the current values. If that's the case, then I'd suggest refactoring and splitting into two Seal functions, one for each case. They can call into a shared helper function to keep the code clean.

Reply:
Done